Unify Pipelines and Firestore classic in CDN and g3 builds

MarkDuckworth opened the pull request:
Autotagged reviewers please skip. Reason: PR to console branch.

For CDN and G3 builds, we will output two bundles:
- firebase-firestore.js - includes only Firestore classic API surface
- firebase-firestore-pipelines.js - includes both the Firestore classic and Firestore Pipelines API surface. The Pipelines API surface is in a namespace called (`pipelines`).

DellaBitta committed:
Grant Create PR content write permissions. (#8848)

Our staging release workflow failed to merge branch. Grant the PR creation workflow `content` `write` permissions to fix this.

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#8849) - v11.5.0

Release version 11.5.0.

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dlarocque committed:
Deprecate `GroundingAttribution` (#8832)

hsubox76 committed:
Stream full logs of test-all firestore job (#8851)

hsubox76 committed:
Set github actions to use commit hashes (#8852)

MarkDuckworth committed:
Markduckworth/ppl mep (#8801)

* Multiple entry points for main package.

* MEP for firestore lite building

* Fix issue with external exports

* Refactor to support multiple entry points for the lite SDK. Fixes import issues and makes code organization across /pipelines and /lite/pipeliens

* Type and comment cleanup

* Make PipelineSource generic and remove useFluentPipelines

* Remove converter and placeholders for converter.

* cleanup: remove documentReferenceFactory

* Linting and formatting

* Update the typings export for pipelines and lite pipelines so that it auguments the Firestore and Query classes.

* Add missing Expr export

* Lint fixes

* TODO cleanup

* update api review

* Query to Proto with integration tests

* API tweaks from review. Selectable, FilterCondition, and Accumulator are abstract classes not interfaces. Removed extraneous Pipeline.sort overload

* Remove console.log statements from prune-dts.ts

* code cleanup in prune-dts

* Add missing imports to pipelines.ts

* If argument to FunctionExpr is a plain object or array, convert these to expression using map(...) or array(...)

* Add tests for evaluating expressions in arrays and maps

* implement new FunctionExpressions as standalone functions

* Fix typo and formatting in pipeline.test.ts

* Implement support for genericStage and genericFunction

* Implement new function expressions as methods on the Expr class.

* Pipeline test reorganization

* Implementing new stages Sample, Union, Unnest, Replace

* Rename FilterCondition to BooleanExpr

* Update functions with variadic params to match signatures of the backend

* Fix and implementation.

* Refactor AggregateFunction out of Expr

* Add ScalarExpr class and make Expr a base class of AggregateFunction

* fixing missing imports and renames

* Remove classes for individual function types

* Separate scalar expressions and aggregate functions into different class hierarchies

* Remove deleted Expr classes from the pipelines exports

* Replace not(fn()) with notFn() in the query to pipeline conversion

* Fix circular dependency between user_data_reader and expressions

* removing modular pipeline creation and removing fluent pipline execution. This was a decision from API review

* Change return type of execute from PipelineResult[] to PipelineSnapshot

* Renames for API review: FunctionExpr, BooleanExpr, params

* missing readonly

* Build and test fixes

* Replace Field.of() with field()

* Add missing export PipelineSnaphsot

* Replace Constant.of() with constant()

* Removed constant(...) overloads to create a map or array. Use the array or map function instead.

* Implemented and documented ascending|descending(fieldName) as a convenient way to create orderings for a specific field

* fix tests creating a constant map and array

* Remove genericFunction, genericAggregateFunction, and genericBooleanExpr. Replace usage with constructor

* Pipeline result timestamp testing

* Testing timestamp edge cases

* Query to pipeline conversion is performed from the PipelineSource class rather than the Query class

* Timestamp tests and fix pipeline tests for change in pipeline creation from a classic query

* test fix

* Refactor and rename all functions to match API proposal changes after API review.

* Fix build issues

* New integration util pipeline_export file to support the pipelines subpackage

* Rename several FunctionExpr standalone exports from <name>Function to <name>. Meets API design

* Removing the all-packages api report script because it is no longer needed

* Remove any from the public api surface

* lint fix

* Code cleanup

* FindNearest integration tests

* Filling in gaps in test coverage

* Implement Pipeline tests for the lite SDK

* Fixed a build issue with the lite SDK

* Improved error message context when invalid user data is passed to a pipeline without being wrapped in a call to constant(...)

* Fix typo in old error message

* Fix lite pipeline exports

* api-review files

* Fixing tests and improving test coverage

* PR feedback

* Updating replaceWith to take an Expr

* yarn format

maneesht committed:
Implement Partial Errors for FDC (#8821)

dlarocque committed:
Add `X-Firebase-AppId` header to VertexAI requests (#8809)

renovate[bot] committed:
Update dependency webpack to v5.94.0 [SECURITY] (#8734)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency express to v4.20.0 [SECURITY] (#8759)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

MarkDuckworth committed:
Port improved whereConditionsFromCursor from Android

MarkDuckworth committed:
Port improved whereConditionsFromCursor from Android (#8867)

dconeybe committed:
karma.base.js: replace "BROWSER" with "BROWSERS" in log messages, to reflect the actual name of the environment variable (see #8491) (#8875)

jamesdaniels committed:
Auth cookie persistence (#8839)

Adding `Persistence.COOKIE` a new persistence method backed by cookies. The `browserCookiePersistence` implementation is designed to be used in conjunction with middleware that ensures both your front and backend authentication state remains synchronized.

google-oss-bot committed:
Merge main into release

dconeybe committed:
firestore: fix `undefined` document snapshot data after "clear site data" (#8871)

dlarocque committed:
Use base browsers variable in Firestore karma config (#8876)

hsubox76 committed:
[chore] Update some dependencies (#8868)

google-oss-bot committed:
Version Packages (#8878)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Merge branch 'release'

milaGGL committed:
Fix 'window is not defined' error when calling clearIndexedDbPersistence

MarkDuckworth committed:
Merge branch 'feat/pipelines' of github.com:firebase/firebase-js-sdk into feat/pipelines

maneesht committed:
Fix Data Connect Types (#8898)

MarkDuckworth committed:
Pipeline and test cleanup

dlarocque committed:
test(vertexai): update mock responses to v8 (#8904)

MarkDuckworth committed:
Test fixes and re-enabling tests for features that were added on backend

MarkDuckworth committed:
Additional test cleanup

MarkDuckworth committed:
Merge commit '88a8055808bdbd1c75011a94d11062460027d931' into feat/pipelines

MarkDuckworth committed:
fix tests and enable only pipeline tests

milaGGL committed:
Fix: display WebChannel error message (#8907)

dlarocque committed:
test(vertexai): add `backendName` param to mock response getters (#8906)

`backendName` can be either `googleAI` or `vertexAI`. This can be passed
to `getMockResponse` or `getMockResponseStreaming` to specify whether
the mock response lookup should be done in the set of vertexAI or
googleAI mock files.

Modified the `convert-mocks.ts` script to read mock responses from the
'developerapi' directory, and add an export to the generated file for
the new lookup object with those mock responses.

milaGGL committed:
fix: remove `null` value inclusion from `!=` and `not-in` filter results (#8915)

MarkDuckworth committed:
Comment cleanup

MarkDuckworth committed:
Add assertion IDs that will be included in production log statements for fail and hardAsserts (#8313)

dlarocque committed:
docs(vertexai): remove HTML `<code>` tags (#8900)

We wrap links that reference public APIs with `<code>` tags so that they
can be rendered as code in devsite. Unfortunately, VSCode's hover
feature can not render these links at all.

Since there are no other alternatives for creating code-formatted links,
we should remove the problematic HTML `<code>` tags.

dlarocque committed:
test(vertexai): update developerapi mock response dir to googleai (#8921)

pashanka committed:
Fix languageCode parameter in action_code_url (#8912)

* Fix languageCode parameter in action_code_url

* Add changeset

google-oss-bot committed:
Merge main into release

dlarocque committed:
test(vertexai): update mock responses to v10 (#8959)

google-oss-bot committed:
Version Packages (#8956)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dlarocque committed:
Merge branch 'release'

M163, 11.6.1

ehsannas committed:
Improve the integration test coverage for online vs offline comparisons. (#8975)

* Improve the integration test coverage for online vs offline comparisons.

* prettier.

* Add missing cases.

dlarocque committed:
fix(vertexai): add missing quote to chat role error message (#8979)

maneesht committed:
Auto Enable SSL for Firebase Studio (#8980)

MarkDuckworth committed:
Cleanup

maneesht committed:
Add support for running the emulators in Cloud Workstation (#8968)

maneesht committed:
Add Cookie Support For Firebase Studio (#8986)

google-oss-bot committed:
Merge main into release

maneesht committed:
Fix Auth Port for Firebase Studio (#8998)

hsubox76 committed:
Change automaticDataCollectionEnabled to default to true. (#8999)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#8997)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Roll back #8999 (#9007)

* Revert "Change automaticDataCollectionEnabled to default to true. (#8999)"

This reverts commit 3789b5ad16ffd462fce1d0b9c2e9ffae373bc6eb.

* Revert #8999 - backend is not ready

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9008)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dlarocque committed:
fix(vertexai): pass `GenerativeModel`'s `BaseParams` to `ChatSession` (#8972)

startChat wasn't passing GenerativeModel's BaseParams to ChatSession.
So, if a model had a generationConfig, it would never be passed to ChatSession.

dlarocque committed:
feat(vertexai): Gemini multimodal output (#8922)

Adds new ResponseModality enum that allows users to specify which modalities should be included in a response.
Since we provide a text() accessor, a similar inlineDataParts() accessor was added to return all InlineDataPart[] in the first candidate.

dlarocque committed:
test: Clean up vscode launch file and add vertexai debug config (#9000)

MarkDuckworth committed:
Fix encoding of CSI keys in Safari (#8993)

hsubox76 committed:
Restore #8999 (#9010)

* Revert "Roll back #8999 (#9007)"

This reverts commit 51e7b489d8aadd531453f882421903da8727b19d.

* Changeset

dlarocque committed:
feat(ai): Migrate to `AI` and add GoogleAI support (#8931)

maneesht committed:
Add Emulator Overlay (#8977)

dlarocque committed:
Rename `@firebase/vertexai` package to `@firebase/ai` (#9011)

hsubox76 committed:
Add Emulator Overlay (#8977) (#9031)

Co-authored-by: Maneesh Tewani <maneesht@users.noreply.github.com>

hsubox76 committed:
add temporary release branch (#9032)

hsubox76 committed:
Allow custom branch for release (#9033)

* add temporary release branch

google-oss-bot committed:
Merge at-11-7-1 into release

hsubox76 committed:
restore Vertex token for temp publish (#9036)

hsubox76 committed:
Version Packages (manual PR) (#9034)

maneesht committed:
Fixed scroll behavior (#9043)

google-oss-bot committed:
Merge at-11-7-1 into release

hsubox76 committed:
Version Packages (manual PR) (#9044)

hsubox76 committed:
Merge branch 'release' into ch-release-merge

hsubox76 committed:
Remove redundant changeset

hsubox76 committed:
Merge pull request #9049 from firebase/ch-release-merge

Merge temporary release branch back into main

hsubox76 committed:
Update e2e tests to reflect AI package and new API (#9017)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9016)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

maneesht committed:
Revert "Fixed scroll behavior (#8977)" (#9053)

* Revert "Fixed scroll behavior (#9043)"

This reverts commit 9bcd1ea9b8cc5b55692765d40df000da8ddef02b.

* Revert "Add Emulator Overlay (#8977) (#9031)"

This reverts commit 8593fa05bd884c2f1f6f3b4ae062efa48af93d24.

* Undid revert of util

* Create dirty-crews-cross.md

* Removed unnecessary markdown file

google-oss-bot committed:
Merge main into release

maneesht committed:
Updated to only show banner when calling connect*Emulator (#9055)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9054)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

MarkDuckworth committed:
Close webchannel instances on terminate (#9041)

Explicitly close any unclosed WebChannel instances on Firestore#terminate()

MarkDuckworth committed:
Reorder reporters to prevent log message duplication in CI (#9040)

dlarocque committed:
docs(ai): Fix 'occured' typo in error codes (#9021)

dlarocque committed:
feat(ai): Add `HarmBlockThreshold.OFF` (#9050)

dlarocque committed:
feat(ai): Schema `title`, `maximum`, `minimum`, and `propertyOrdering` (#9047)

dlarocque committed:
feat(ai): Add support for `minItems` and `maxItems` to `Schema` (#9026)

hsubox76 committed:
Build out e2e subdirectories (#9065)

maneesht committed:
Added test project to fdc (#8924)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9073)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Release version 11.9.0.

dlarocque committed:
test(ai): add integration tests (#8853)

hsubox76 committed:
Update issue template to rename VertexAI > AI (#9074)

hsubox76 committed:
ci: Fix Firefox error and streamline cross-browser tests (#9015)

maneesht committed:
Fixed issue where requestSts wasn't including the Studio cookie in it (#9075)

maneesht committed:
Fix issue where we ping an endpoint that hasn't been implemented (#9059)

maneesht committed:
Updated firebase auth changeset (#9088)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
build(all): Version Packages v11.9.1 (#9089)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

v11.9.1 release.

dlarocque committed:
test(ai): Upgrade mock responses to v14 (#9091)

MarkDuckworth committed:
Build and test fixes

eranc-google committed:
Feat/google3 typings support (#9085)

dlarocque committed:
fix(ai): Add deprecation tag to `totalBillableCharacters` (#9090)

dlarocque committed:
test(ai): Add `gemini-2.5-flash` to integration tests (#9110)

Added gemini-2.5-flash to our integration tests now that it's publicly available.

The token counts differ slightly between 2.0-flash and 2.5-flash so I introduced conditionals when checking token counts.

dconeybe committed:
firestore: minor refactor of listener registration of "versionchange" indexedb events (#9087)

maneesht committed:
Fixed issue where storage wasn't passing cookies when using firebase Studio (#9111)

hsubox76 committed:
fix: Add "react-native" entry point to @firebase/app (#9112)

DellaBitta committed:
Feat(Firestore) JSON serialization of types to improve SSR support. (#8926)

Support the ability to resume `onSnapshot` listeners in the CSR phase based on serialized `DataSnapshot`s and `QuerySnapshot`s built in the SSR phase. Allow Firestore result types to be serialized with `toJSON` and then deserialized with `fromJSON` methods on the objects.

`DocumentSnapshot` and `QuerySnapshot` deserialization methods will be standalone, tree-shakable functions `dataSnapshotFromJSON` and `querySnapshotFromJSON`.

DellaBitta committed:
chore(Firestore) add changeset for PR #8926 (#9115)

Add the missing changset for PR #8926.

google-oss-bot committed:
Merge main into release

raymondlam committed:
Update Fireperf logging to use sendBeacon only if the payload is under the 64KB limit (#9120)

* Update Fireperf logging to use sendBeacon only if the payload is under the 64KB limit for most browsers.

- For the flush, attempt to use sendBeacon with a low number of events incase sendBeacon is also used by other libraries.

* Add changeset and fix format

* Add additional comments

* Put max flush size behind remote config flag

MarkDuckworth committed:
Merge branch 'main' of github.com:firebase/firebase-js-sdk into feat/pipelines

google-oss-bot committed:
Version Packages (#9126)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

mansisampat committed:
Increase unit test coverage for facebook.ts to 100% (#9129)

* Increase unit test coverage for facebook.ts to 100%

mansisampat committed:
Increase unit test coverage for github.ts to 100% (#9130)

* Increase unit test coverage for github.ts to 100%

dlarocque committed:
test: Use inline source maps for webpack test output (#9109)

mansisampat committed:
Increase unit test coverage for saml.ts to 100% (#9131)

* Increase unit test coverage for saml.ts to 100%

dconeybe committed:
firestore: misc.ts: further improved performance of UTF-8 string comparison logic

hsubox76 committed:
fix(ai): typings for functionDeclaration.parameters. (#8948)

dlarocque committed:
feat(ai): add support for grounding with google search (#9068)

dlarocque committed:
feat!(ai): Remove `vertexai` import alias (#9081)

dconeybe committed:
docs(firestore): add an example to the detailed comment in compareUtf8Strings() (#9154)

dlarocque committed:
feat(ai): Add support for Thinking Budget (#9156)

dlarocque committed:
feat!(ai): remove `VertexAI` APIs (#9063)

hsubox76 committed:
fix!(ai): convert TS enums exports in Firebase AI into const variables (#9137)

hsubox76 committed:
Merge branch 'release'

dconeybe committed:
fix(firestore): revert fix for `undefined` document snapshot data after "clear site data" (#9162)

This reverts https://github.com/firebase/firebase-js-sdk/pull/8871 because it had unexpected negative customer impacts (https://github.com/firebase/firebase-js-sdk/issues/9056) and caused CI failures.

hsubox76 committed:
build!: update build targets to ES2020 (#9101)

DellaBitta committed:
(feat) initializeServerApp support for App Hosting auto init (#9151)

Implement Auto Init for `initializeServerApp`. 

Auto init was previously implemented for `initializeApp` in #8483. This PR adds the same functionality to the `initializeServerApp` API surface.

Fixes #8863

dlarocque committed:
feat!(ai): Remove `GroundingAttribution` (#9079)

dlarocque committed:
feat!(ai): add support for `anyOf` schemas (#9042)

hsubox76 committed:
chore!: update engines.node to minimum of 20 (#9128)

hsubox76 committed:
fix(ci): Update e2e tests to import from ai instead of vertexai (#9164)

google-oss-bot committed:
Merge main into release

hsubox76 committed:
chore: Add missing changeset for 9101 (#9167)

google-oss-bot committed:
Merge main into release

MarkDuckworth committed:
Revert "firestore: minor refactor of listener registration of "versionchange" indexedb events (#9087) (#9168)

* Revert "firestore: minor refactor of listener registration of "versionchange" indexedb events (#9087)"

This reverts commit f73e08b212314547b39a10cd3e393f9e94776f21.

* Create hungry-icons-dream.md

google-oss-bot committed:
Merge main into release

hsubox76 committed:
docs: fix doc typos found by g3 presubmit (#9166)

google-oss-bot committed:
Version Packages (#9165)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Merge branch 'release'
Release 12.0.0

raymondlam committed:
Fixes a bug where the out-of-the-box core web vital metrics would throw an error when the target element name exceeds the max length of a custom attribute value. (#9178)

nicole0707 committed:
fix(app-check): Prevent redundant exchangeToken calls in debug mode (#9187)

dlarocque committed:
Update BrowsersList DB (#9188)

gsiddh committed:
Firebase AI Hybrid Inference Implementation (#9029)

Co-authored-by: Daniel La Rocque <dlarocque@google.com>
Co-authored-by: Erik Eldridge <erikeldridge@google.com>
Co-authored-by: Christina Holland <chholland@google.com>

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9195)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Backfill fixes for hybrid docs (#9197)

maneesht committed:
Fix Firebase Functions Emulator usage on Firebase Studio (#9204)

* Used valid path for pingserver

* Added test

* Create three-balloons-collect.md

* Fixed fromUrl

* Removed only

* Fixed typings

cherylEnkidu committed:
fix(firestore): fix empty message reject inside transaction body (#9177)

hsubox76 committed:
Add limitedUseToken option to AI SDK (#9201)

andrewheard committed:
[Firebase AI] Add thought summary and signature support (#9192)

dlarocque committed:
fix(ai): Only include unexpected `prediction` element in error message (#9222)

* fix(ai): Only include unexpected prediction element in error message.

If the `responseJson` from the Predict request has an image, the entire
image base64 will be included in the error message. This can make the error
message >400kB (!!), which could cause performance issues in apps.

I ran into this when testing https://github.com/firebase/firebase-js-sdk/pull/9216,
since the `safetyAttributes` would be included in a response and trigger
this error, but the `predictions` array in `responseJson` would have images.
I noticed that when the sample app rendered the `error.message`, it
began running into serious performance issues.

dlarocque committed:
fix(ai): Add `'includeSafetyAttributes'` to Predict request payloads (#9216)

hsubox76 committed:
fix(ai): keep window-specific code out of Node bundle (#9225)

dlarocque committed:
feat(ai): add support for the Live API (#9224)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9230)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
remove accidental factory export (#9232)

google-oss-bot committed:
Merge main into release

google-oss-bot committed:
Version Packages (#9233)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
fix(ai): Fix test script broken by #9232 (#9235)

daymxn committed:
Update CHANGELOG.md (#9237)

hsubox76 committed:
feat(ai): Add prefer_in_cloud option for inference mode (#9236)

Co-authored-by: Daniel La Rocque <dlarocque@google.com>
Co-authored-by: rachelsaunders <52258509+rachelsaunders@users.noreply.github.com>

tusharkhandelwal8 committed:
Remote Config Realtime for Web (#9221)

hsubox76 committed:
feat(ai): Add code execution (#9249)

MarkDuckworth committed:
chore(firestore): Update closure-net dependency to the latest release (#9242)

* Update closure-net dependency to the latest release (6f48f578d3e80fe7a85e530a5d95b9351433d135)

hsubox76 committed:
Change documentation tags for hybrid inference from EXPERIMENTAL to public preview (#9253)

google-oss-bot committed:
Merge main into release

hsubox76 committed:
docs: Fix docgen script to clean out package temp folders first (#9250)

google-oss-bot committed:
Version Packages (#9258)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Merge branch 'release'
Release 12.3.0

hsubox76 committed:
Expose setUserProperties from internal analytics instance (#9263)

dlarocque committed:
feat(ai): add support for URL context (#9254)

hsubox76 committed:
docs(ai): tag code execution with "beta" tag (public preview) (#9274)

hsubox76 committed:
fix(ai): Handle empty parts when streaming (#9262)

dlarocque committed:
feat(ai): Add method to send function responses in a live session (#9272)

hsubox76 committed:
Update ai model name used for e2e tests (#9282)

MarkDuckworth committed:
Merge branch 'main' of github.com:firebase/firebase-js-sdk into feat/pipelines

dlarocque committed:
docs(ai): promote gemini developer api and imagen to GA (#9255)

google-oss-bot committed:
Merge main into release

MarkDuckworth committed:
Implementing features from API audit and node backport (#9284)

google-oss-bot committed:
Version Packages (#9300) release v12.4.0

For JS SDK release v12.4.0

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

MarkDuckworth committed:
Fix pipelines CI (#9306)

* Skip pipeline tests in lite, for now

* update bundle definitions for size report

* Fix size testing for pipelines bundles

* fixed typings

* Fixed package paths

* Fix integration tests that run on the built packages

MarkDuckworth committed:
only run pipeline tests when pointing to the enterprise db

MarkDuckworth committed:
fix typings:public script for firestore

Navya-arugonda committed:
fix(auth): export MISSING_PASSWORD in AuthErrorCodes (#9297)

MarkDuckworth committed:
dist/firestore/src/global_index.d.ts now includes pipelines in namespace 'pipelines'

MarkDuckworth committed:
Update global.ts

MarkDuckworth committed:
Custom replacer plugin to remove declare module block from global.d.ts

hsubox76 committed:
feat(ai): make generateContent methods return inferenceSource (#9315)

MarkDuckworth committed:
Update CDN and g3 builds of Firestore to include Pipelines in the firebase-firestore.js file

MarkDuckworth committed:
Updated CDN bundles so that firebase-firestore.js is not modified, but firebase-firestore-pipelines.js contains both classic and pipelines APIs.

MarkDuckworth committed:
Fix regression in _internalPipelineToExecutePipelineRequestProto

MarkDuckworth committed:
Fixing missing paths in package.json

MarkDuckworth committed:
Fix binary size script

MarkDuckworth committed:
Add support for index.cdn.ts files in packages/firebase. Then added a cdn.ts to packages/firebase/firestore/pipelines

MarkDuckworth committed:
send api key to DBE requests (#9331)

MarkDuckworth committed:
send api key to DBE requests (#9331)

hsubox76 committed:
Fix logic for merging default onDeviceParams with user-provided onDeviceParams (#9314)

hsubox76 committed:
feat(auth): export beta tagged symbols to public typings (#9313)

dlarocque committed:
feat(ai): Add `LiveSession` realtime APIs (#9291)

MarkDuckworth committed:
Rename grpc_connection.node.test.ts, so the file is filtered from browser unit testing.

dlarocque committed:
test(ai): Update mock responses to v15 (#9328)

dlarocque commented on packages/firestore/rollup.config.js:
```diff
@@ -59,6 +59,35 @@ const browserPlugins = [
 ];
 
+// TODO - update the implementation to match all content in the declare module block.
+function declareModuleReplacePlugin() {
```
All our custom rollup plugins exist in `scripts/build`. Can you move this plugin to a new file in that directory?

MarkDuckworth replied:
moved to `scripts/build`. Thanks.

dlarocque commented on packages/firestore/src/core/firestore_client.ts:
```diff
@@ -146,7 +146,11 @@ export class FirestoreClient {
     public asyncQueue: AsyncQueue,
-    private databaseInfo: DatabaseInfo,
+    /**
+     * @internal
```
I don't think `@internal` is necessary here, since `FirestoreClient` is not a public API.

MarkDuckworth committed:
backport fixes in the console to feat/pipelines (#9340)

* dist/firestore/src/global_index.d.ts now includes pipelines in namespace 'pipelines'

* Update global.ts

* Custom replacer plugin to remove declare module block from global.d.ts

* Update CDN and g3 builds of Firestore to include Pipelines in the firebase-firestore.js file

* Updated CDN bundles so that firebase-firestore.js is not modified, but firebase-firestore-pipelines.js contains both classic and pipelines APIs.

* Fix regression in _internalPipelineToExecutePipelineRequestProto

* Fixing missing paths in package.json

* Fix binary size script

* Add support for index.cdn.ts files in packages/firebase. Then added a cdn.ts to packages/firebase/firestore/pipelines

* Rename grpc_connection.node.test.ts, so the file is filtered from browser unit testing.

dlarocque committed:
feat(ai): Add Live API transcriptions (#9330)

MarkDuckworth committed:
Throw on duplicate alias (#9343)

MarkDuckworth committed:
Pipeline fix tests and exports (#9342)

* clean up exports and imports

* Fix implementation of Query to Pipeline for isNan, isNotNan, isNull, and isNotNull changes in the backend

* fix tests to accomodate for backend changes to multiple sort stages

* another test that the console usage will perform runtime validation of user data

* formatting

* Copy the latest pipelines tests into lite api pipelines tests. Fix a few minor bugs in the lite pipeline implementation

* Remove special handling of null and nan in query to pipeline

* remove function expresssions isNan, isNull, isNotNan, isNotNull, and error

* Renamed len function to length, with a fix to the prune-dts script

* lastest API review md files

* fixed title is not null test

google-oss-bot committed:
Merge main into release

hsubox76 committed:
test(ai): Fix node tests (#9350)

MarkDuckworth committed:
Add support for sending the API key for web channel streams (#9354)

MarkDuckworth committed:
Merge branch 'main' of github.com:firebase/firebase-js-sdk into feat/pipelines

MarkDuckworth committed:
Merge branch 'feat/pipelines' into markduckworth/g3-import-firestore-globals

MarkDuckworth committed:
fixes

MarkDuckworth committed:
add beta tags to pipelines api. Add missing tsdoc comments

MarkDuckworth committed:
Yarn format

google-oss-bot committed:
Version Packages (#9353)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dlarocque committed:
Merge branch 'release'

maneesht committed:
Fixed onComplete for FDC (#9349)

MarkDuckworth committed:
add functions timestampTrunc, split, and type

MarkDuckworth committed:
Add valueToTrim parameter to trim

MarkDuckworth committed:
pass value to trim

MarkDuckworth committed:
Add Bytes to arg type for trim

MarkDuckworth committed:
clean up integration tests so they pass in classic and enterprise DBs

MarkDuckworth committed:
Merge branch 'main' of github.com:firebase/firebase-js-sdk into feat/pipelines

MarkDuckworth committed:
Missing file

MarkDuckworth committed:
cleanup

MarkDuckworth committed:
Fix api review files and missing export TimeGranularity

dlarocque committed:
test(ai): do not run browser tests when debugging node tests (#9363)

MarkDuckworth committed:
fixed private constructor on aggregatefunction

MarkDuckworth committed:
Add _fieldsProto() method to snapshot and result classes for console support

MarkDuckworth committed:
Merge branch 'feat/pipelines' into markduckworth/g3-import-firestore-globals

athiramanu committed:
feat(rc): Web support for ABT & Rollouts (#9293)

Provide experiment in details in fetch response and integrate with Google Analytics to create experiment related user properties

dlarocque committed:
feat(ai): server prompt templates (#9351)

hsubox76 committed:
Revert "feat(rc): Web support for ABT & Rollouts (#9293)" (#9374)

This reverts commit 6abe52967f3d1ca4a02d4469a14c6d692e1f3b31.

google-oss-bot committed:
Merge main into release

hsubox76 committed:
Fix functions deploy by upgrading to node 20 (#9377)

google-oss-bot committed:
Merge main into release

dlarocque committed:
test(ai): update vertex ai live model name (#9376)

google-oss-bot committed:
Version Packages (#9375)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hsubox76 committed:
Merge branch 'release'
Release 12.6.0

MarkDuckworth committed:
Refactoring rollup plugin to extract the contents of the database module declaration in global_index.d.ts. This is responding to feedback from the peer review about location and documentation of this plugin

MarkDuckworth committed:
Exporting proto types that are access directly by the console.

MarkDuckworth committed:
Release 12.6.0 - Merge branch 'main' of github.com:firebase/firebase-js-sdk into feat/pipelines

MarkDuckworth committed:
Merge branch 'feat/pipelines' into markduckworth/g3-import-firestore-globals. Gets release 12.6.0 from main.

MarkDuckworth committed:
fixed regex in rollup plugin

MarkDuckworth committed:
add missing copyright header
